Installation fails on latest Alpine 3.19.0

```
In file included from absl/base/internal/low_level_alloc.cc:26:
../src/absl/base/internal/direct_mmap.h:75:25: error: 'off64_t' has not been declared
   75 |                         off64_t offset) noexcept {
      |                         ^~~~~~~
```

See https://github.com/abseil/abseil-cpp/issues/1473 for the corresponding upstream issue.

This can be fixed by porting Commit https://github.com/abseil/abseil-cpp/commit/4500c2fada4e952037c59bd65e8be1ba0b29f21e from upstream. I have confirmed that **s2** 1.1.4 installs and checks successfully on Alpine 3.19.0 if that patch is applied.

--- a/src/absl/base/internal/direct_mmap.h
+++ b/src/absl/base/internal/direct_mmap.h
@@ -72,7 +72,7 @@ namespace base_internal {
 // Platform specific logic extracted from
 // https://chromium.googlesource.com/linux-syscall-support/+/master/linux_syscall_support.h
 inline void* DirectMmap(void* start, size_t length, int prot, int flags, int fd,
-                        off64_t offset) noexcept {
+                        off_t offset) noexcept {
 #if defined(__i386__) || defined(__ARM_ARCH_3__) || defined(__ARM_EABI__) || \
     defined(__m68k__) || defined(__sh__) ||                                  \
     (defined(__hppa__) && !defined(__LP64__)) ||                             \
@@ -101,7 +101,7 @@ inline void* DirectMmap(void* start, size_t length, int prot, int flags, int fd,
 #else
   return reinterpret_cast<void*>(
       syscall(SYS_mmap2, start, length, prot, flags, fd,
-              static_cast<off_t>(offset / pagesize)));
+              static_cast<unsigned long>(offset / pagesize)));
 #endif
 #elif defined(__s390x__)
   // On s390x, mmap() arguments are passed in memory.

